proto/viam/app/mltraining/v1/ml_training.proto

```diff
@@ -68,6 +69,7 @@ message TrainingJobMetadata {
   string synced_model_id = 5 [(tagger.v1.tags) = "bson:\"synced_model_id\" json:\"synced_model_id\""];
   string user_email = 6 [(tagger.v1.tags) = "bson:\"user_email\" json:\"user_email\""];
   string id = 7 [(tagger.v1.tags) = "bson:\"_id\" json:\"id,omitempty\""];
+  string error_msg = 8 [(tagger.v1.tags) = "bson:\"error_msg\" json:\"error_msg\""];
```

Comment:
[supernit]: Sorry for the back and forth on this, but given that we don't know the exact details of how we will implement this (e.g. do we also want status codes?), I am leaning towards leaving this out of this PR and including it when we actually do the error message, which if IIUC will come post-implementation